### PR TITLE
Fix personalization test and add check for optionalProvider.

### PR DESCRIPTION
**Pull request by rlazo**
The personalization test was using the default namespace, and because another instance of config already existed in that namespace, the test instance was not being created. This caused the personalization test to try to use real objects instead of the mocks, including the direct executor. This PR fixes this and uses a provider to fake the interaction when Analytics is present and when it's not.

### CODE REVIEW

File: firebase-config/src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigTest.java
Comment (vkryachko) [inline firebase-config/src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigTest.java:1003]: TIL, cool library!

**Review by vkryachko — COMMENTED** [review]
I wonder if the perf test failure a flake or is caused by the change in RC

**Comment by rlazo** [comment]
> I wonder if the perf test failure a flake or is caused by the change in RC

Flake

**Review by vkryachko — APPROVED** [review]
LGTM, but let's wait for @danasilver to review before merging.

**Comment by rlazo** [comment]
SGTM

**Review by danasilver — APPROVED** [review]

**Pull request merged**
